clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy

```diff
@@ -19,31 +19,99 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
 import java.util.regex.Matcher
 import java.util.regex.Pattern
 
+/**
+ * Load balancer description is a comma-deliminated set
+ * of key-value pairs.
+ *
+ * Usually this will be in the form of "internal_port=8100,created_time=12345678
+ */
 trait LoadBalancerResolver {
 
   final String nameRegex = "(\\w+)-(\\w+)-(\\w+)"
   final Pattern namePattern = Pattern.compile(nameRegex)
-  final String descriptionRegex = ".*internal_port=([0-9]+).*"
-  final Pattern descriptionPattern = Pattern.compile(descriptionRegex)
+  final String portRegex = ".*internal_port=([0-9]+).*"
+  final Pattern portPattern = Pattern.compile(portRegex)
+  final String createdRegex = ".*created_time=([0-9]+).*"
+  final Pattern createdPattern = Pattern.compile(createdRegex)
 
   String getBaseName(final String derivedName) {
     String result = null
     if (derivedName) {
       Matcher matcher = namePattern.matcher(derivedName)
-
       if (matcher.matches() && matcher.groupCount() == 3) {
         result = matcher.group(1)
       }
     }
     result
   }
 
+  /**
+   * Parse the internal port from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,internal_port=8100,...
+   * }
+   * @param description
+   * @return the port value
+   */
   int getInternalPort(final String description) {
-    int result = -1
-    if (description) {
-      Matcher matcher = descriptionPattern.matcher(description)
+    String s = match(description, portPattern)
+    if (s) {
+      s.toInteger()
+    } else {
+      -1
+    }
+  }
+
+  /**
+   *
+   * @param port
+   * @return
+   */
+  String generateInternalPort(int port) {
+    "internal_port=${port}"
+  }
+
+  /**
+   * Parse the created time from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,created_time=12345678,...
+   * }
+   * @param description
+   * @return the port value
+   */
+  long getCreatedTime(final String description) {
```

Comment:
I wonder if the `setCreatedTime` and `setInternalPort` should go in this class as well